### PR TITLE
Add basic support for converting CAG into XLSX

### PR DESCRIPTION
**Pull request by Xarthisius**
### How to test?

1. Grab [AAB06 Auto Analysis.cag](https://drive.google.com/drive/folders/16miKqlYsAsJCRd3mi5jYcqlhmnrgFA7s) and save it as `AAB06.cag`
2. Run `htmdec-formats cag-to-xlsx AAB06.cag AAB06.xlsx`
3. Compare the result with `AAB06 Analysis_Volume & area.xlsx` (same folder as in 1.)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,9 +27,10 @@ BROWSER := python -c "$$BROWSER_PYSCRIPT"
 help:
 	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
 
-kaitai: src/htmdec_formats/ksy_files/nmdfile.py src/htmdec_formats/ksy_files/simple_xls.py ## rebuild kaitai .py files from .ksy files
+KSY_SOURCES := $(shell find src/htmdec_formats/ksy_files -name '*.ksy')
+COMPILED_KSY := $(KSY_SOURCES:%.ksy=%.py)
 
-%.py : %.ksy
+src/htmdec_formats/ksy_files/%.py : src/htmdec_formats/ksy_files/%.ksy
```
Comment (matthewturk) [inline Makefile:33]: nice!

File: src/htmdec_formats/keyence_formats.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        self.meta_files = {}
+        self.volumes = {}
+        self.measurements = {}
+        self.vk6_files = {}
+
+        # CAG uses a lot of uuid instead of human readable names
+        # following mapping is used to map uuid to something more reasonable.
+        # Note that names are made up.
+        self.fmapping = {
+            "f1724dc6-686c-4502-9227-2a594bc8ed33": "index.xml",
```
Comment (matthewturk) [inline src/htmdec_formats/keyence_formats.py:35]: names are made up, or uuids are made up?  Are you sure that these UUIDs are general?

**Review by matthewturk — COMMENTED** [review]
looks great.  Did you make any changes to my KSY file for dotnet?

File: src/htmdec_formats/keyence_formats.py (hunk 2 above)
Comment (Xarthisius) [reply src/htmdec_formats/keyence_formats.py:35]: names are made up. UUIDs were the same in every file I checked.

**Comment by Xarthisius** [comment]
> looks great. Did you make any changes to my KSY file for dotnet?

No, I think it's exactly the same thing you gave me.

**Pull request merged**